src/SwedbankPay/Core/Library/PaymentInfo.php

```diff
@@ -84,13 +88,55 @@ public function request($method, $url, $params = [])
                     }
                 }
 
-                throw new Exception($message, 0, null, $data['problems']);
+                throw new Exception($message, $httpCode, null, $data['problems']);
             }
 
             throw new Exception('API Exception. Please check logs');
         }
     }
 
+	/**
+	 * Check API Credentials.
+	 *
+	 * @return bool Returns true if ok
+	 * @throws Exception
+	 */
+	public function checkApiCredentials()
+	{
+		/**
+		 * Perform a POST request to create a payment or paymentorder with invalid data that is known to fail with
+		 * 400 Bad Request.
+		 * This is to ensure that no payments or payment orders are actually created if the request succeeds.
+		 * If the request fails with 401, something is wrong with the credentials.
+		 * If the request fails with 403 something is wrong with the contract.
+		 * If the request fails with 400 Bad Request, credentials and contracts should be OK.
+		 */
+
+		$params = [
+			'payment' => [
+				'operation' => 'Test',
+				'payeeInfo' => [
+					'payeeId' => $this->getConfiguration()->getPayeeId(),
+					'payeeName' => $this->getConfiguration()->getPayeeName(),
+				]
+			]
+		];
+
+		try {
+			$this->request('POST', '/psp/creditcard/payments', $params);
```

Comment:
We don't know that the customer has access to the `creditcard` instrument. I think we need one `checkApiCredentials()` method per payment instrument, or one method that takes in `paymentInstrument` as a parameter.